package.json

```diff
@@ -12,9 +12,9 @@
     "capi"
   ],
   "dependencies": {
-    "aws-amplify": "latest",
-    "@aws-amplify/ui-components": "latest",
+    "@aws-amplify/ui-components": "^1.6.2",
     "array-flatten": "^3.0.0",
+    "aws-amplify": "^4.2.0",
```

Comment:
I believe that `latest` is correct because we want our docs to use the latest artifacts & published code.  Even with `latest`, we can see that a very old version was "locked" here:
https://github.com/aws-amplify/docs/pull/3415/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deL147

So the problem appears to be `yarn.lock` doing what it's intended to do – deterministic installs.

I don't know the best way to tell `yarn` or `npm` "yes, this is a dependency, but don't lock the version because we always want the `latest` from NPM".

Technically, that'd make these `peerDependencies`, but then that means both contributors and CI would need to `yarn install aws-amplify @aws-amplify/ui-components` before running the project.

Reply:
Só how to use latest and get Yarn to behave correctly and update `yarn.lock`?

Reply:
In the [amplify prebuild phase](https://github.com/aws-amplify/docs/blob/5c17716ca465d0e3dab64f8e3573c4e06f79feaa/amplify.yml#L4-L6), could we add the [`upgrade --latest`](https://classic.yarnpkg.com/en/docs/cli/upgrade/#toc-yarn-upgrade-package-latest-l-caret-tilde-exact-pattern) flag for `aws-amplify` and `@aws-amplify/ui-components`?

This doesn't solve the issue for local development, but I think this would at least allow the deployed site to always be get the most recent version.

```
yarn upgrade aws-amplify @aws-amplify/ui-components --latest
```

Reply:
I understand the intent of always pointing to the latest `amplify` in the docs, but it seems like `latest` is not giving us what we need. The previous config remained outdated almost a year, which is a concerning outcome. So I agree with the premise but it doesn't seem like keeping `latest` is correct unless we can adapt the development and the CI/CD process to get the result we intend.

@mauerbac Do you have any thoughts on this? It seems like there's a missing link between the Amplify release process and the Docs dependency on it.

Reply:
@drochetti  @ericclemmons  What if we add a [`dependabot config`](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates) so that dependabot opens a docs PR when `amplify` updates?

<img width="982" alt="image" src="https://user-images.githubusercontent.com/2081699/127041567-98cf7f57-bf62-4302-ad84-f5348bda5e89.png">

Reply:
@drochetti @ericclemmons I've opened a PR to add a dependabot config for this:
https://github.com/aws-amplify/docs/pull/3607